src/site/twitch.tv/modules/settings/SettingsMenuButton.vue

```diff
@@ -11,37 +11,103 @@
+
+const hideBitsButton = useConfig<boolean>("layout.hide_bits_buttons");
+const hideSiteNotificationsButton = useConfig<boolean>("layout.hide_site_notifications_button");
+const hidePrimeCrown = useConfig<boolean>("layout.hide_prime_offers");
+const hideTurboButton = useConfig<boolean>("layout.hide_turbo_button");
+const hideWhispersButton = useConfig<boolean>("layout.hide_whispers_button");
+
+const menuElements = reactive({
+	renderBitsButton: hideBitsButton,
+	renderOnsiteNotificatons: hideSiteNotificationsButton,
+	renderTwitchPrimeCrown: hidePrimeCrown,
+	renderTurboButton: hideTurboButton,
+	renderWhispers: hideWhispersButton,
+});
+
+// force update top nav when user changes settings
+watch(menuElements, (renderElements) => {
+	rerenderTopNav(renderElements);
+});
+
+// TODO: define proper types here
+function rerenderTopNav(els) {
+	if (!props.inst?.component) return;
+	for (const renderFunc in els) {
+		defineFunctionHook(props.inst.component, renderFunc, function (old, ...args: unknown[]) {
+			return els[renderFunc] ? null : old?.apply(this, args);
+		});
+	}
+	toRaw(props.inst.component).forceUpdate();
+}
 
 const el = document.createElement("div");
 el.id = "seventv-settings-button";
 const containerEl = ref<HTMLElement>(el);
 
-// Topnav on the main twitch page
-let menuButtons = document.querySelector(".top-nav__menu")?.lastChild;
 let tooltip = "under";
+function insert7tvButton() {
+	const rootNode = props.inst?.domNodes.root;
+	let navmenu;
+	if (rootNode) {
+		// Place 7tv button next to the user profile
+		navmenu = rootNode.querySelector(".top-nav__menu")?.lastChild;
+	} else {
+		// Legacy
+		// Topnav on the main twitch page
+		navmenu = document.querySelector(".top-nav__menu")?.lastChild;
 
-if (!menuButtons) {
-	// Sidenav in modview
-	menuButtons = document.querySelector(".modview-dock__followed-channels")?.nextSibling;
-	tooltip = "right";
-}
-if (!menuButtons) {
-	// Header if popout
-	menuButtons = document.querySelector(".stream-chat-header")?.lastChild;
-	tooltip = "left";
-}
-if (menuButtons) {
-	menuButtons.insertBefore(containerEl.value, menuButtons.lastChild);
+		if (!navmenu) {
+			// Sidenav in modview
+			navmenu = document.querySelector(".modview-dock__followed-channels")?.nextSibling;
+			tooltip = "right";
+		}
+		if (!navmenu) {
+			// Header if popout
+			navmenu = document.querySelector(".stream-chat-header")?.lastChild;
+			tooltip = "left";
+		}
+	}
+	if (!navmenu) return;
+	if (navmenu.contains(containerEl.value)) {
+		// we have to re-insert the 7tv button since re-rendering will change order
+		navmenu.parentElement?.querySelector<HTMLElement>("#seventv-settings-button")?.remove();
+		navmenu.insertBefore(containerEl.value, navmenu.lastChild);
+	} else {
+		navmenu.insertBefore(containerEl.value, navmenu.lastChild);
+	}
 }
+
+// we need to render the current state from the user settings
+// this should only run once after mount
+watch(
+	() => props.inst,
+	() => {
+		rerenderTopNav(menuElements);
```

Comment:
This is a bit non descriptive as a function, you should probably do the checks for existence here so its a bit more clear whats going on.